packages/python/accern_xyme/accern_xyme.py

```diff
@@ -1849,6 +1860,46 @@ def delete(self) -> DeleteBlobResponse:
             },
         ))
 
+    def download_full_dag_zip(
+            self, to_path: Optional[str]) -> Optional[io.BytesIO]:
+        cur_res, _ = self._client.request_bytes(
+            METHOD_GET, "/download_dag_zip", {
+                "dag": self.get_uri(),
+            })
+        if to_path is None:
+            return io.BytesIO(cur_res.read())
+        with open(to_path, "wb") as file_download:
+            file_download.write(cur_res.read())
+        return None
+
+    def _upload_dag_blobs(
+            self,
+            tmpdir: str,
+            blobs_map: Dict[str, str]) -> List['BlobHandle']:
+        blob_handles = []
+        for blob_uri, blob_fname in blobs_map.items():
+            blob_file = os.path.join(tmpdir, blob_fname)
+            blob_handle = self._client.get_blob_handle(blob_uri)
+            blob_handles.extend(blob_handle.upload_zip(blob_file))
+        return blob_handles
+
+    def upload_full_dag_zip(self, source: str) -> List['BlobHandle']:
+        dag_blob_handle = self._client.get_blob_handle(self.get_uri())
+        blob_handles = []
+        with tempfile.TemporaryDirectory() as tmpdir:
+            shutil.unpack_archive(source, tmpdir, "zip")
+            fmap_json = os.path.join(tmpdir, "filemap.json")
+            with open(fmap_json, "r") as fmap:
```

Comment:
use TypedDict
then use dag_fname to upload first file
then loop through blobs
you'll always get both uri and fname
check if fname == dag_fname for skipping

Reply:
don't skip